test/E2E/Azure.Functions.PowerShellWorker.E2E/Azure.Functions.PowerShellWorker.E2E/ManagedDependenciesEndToEndTests.cs

```diff
@@ -14,7 +15,7 @@ public async Task ManagedDependenciesAreAvailable()
         {
             var actualResponseMessage =
                 await Utilities.InvokeHttpTrigger("UsingManagedDependencies", string.Empty, HttpStatusCode.OK);
-            Assert.Contains(@"\ManagedDependencies\Az\", actualResponseMessage);
+            Assert.Matches(new Regex(@"[\\/]ManagedDependencies[\\/]Az[\\/]"), actualResponseMessage);
```

Comment:
Regex works, but we could also use ```[io.path]::Combine("ManagedDependencies", "Az")```.

Reply:
I agree, but I was also thinking about the next step: for https://github.com/Azure/azure-functions-powershell-worker/pull/267, we'll need to expect something like `...\ManagedDependencies\{unique id}\Az\...`, so simply combining known parts will not work anymore. We'll have to either split the path and compare separate components, or just update the regex to something like `[\\/]ManagedDependencies[\\/]([^\\/]+?)[\\/]Az[\\/]`. Not that this one is extremely readable though... Ok, let me think about it.

Reply:
You can use:

```
System.IO.Path.DirectorySeparatorChar
```

which will be / on non-Windows and \ on Windows

Reply:
Sure, but we'll still need to handle the unknown `{unique id}` part somehow (not right now, but soon). Using `System.IO.Path.DirectorySeparatorChar` in a regex will probably lead to something like: `new Regex($"\{DirectorySeparatorChar}ManagedDependencies\{DirectorySeparatorChar}(^\{DirectorySeparatorChar}+?)\{DirectorySeparatorChar}Az\{DirectorySeparatorChar}")`, which makes me a bit dizzy... :-)

Alternatively, we forget the regex, split the path into components, find the position of "ManagedDependencies" and expect "Az" at the N+2 position. But the nice thing about `Assert.Matches` is that it is a one-liner that automatically prints the entire actual string on failures.

Reply:
IMO, I like the latter :) more readable.

Reply:
I like the the last option as well :-).

Reply:
Ok. Let's do this: for this PR, I will do `[io.path]::Combine("ManagedDependencies", "Az")`. For https://github.com/Azure/azure-functions-powershell-worker/pull/267, I will do splitting etc. (unless we have a better idea by then).